Form/ChoiceList/ModelChoiceLoader.php

```diff
@@ -109,7 +110,7 @@ public function loadChoiceList($value = null)
                     }
                 }
 
-                $id = implode('~', $this->getIdentifierValues($entity));
+                $id = implode(AdapterInterface::ID_SEPARATOR, $this->getIdentifierValues($entity));
```

Comment:
See const [AdapterInterface::ID_SEPERATOR](https://github.com/sonata-project/SonataCoreBundle/blob/3.x/Model/Adapter/AdapterInterface.php#L16).
I am faced with the fact that the composite primary key is divided character `~`. I suggested that to separate the primary key used by this constant. I could be wrong on that score.